# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/ContentIterator.h
# Commit: dda3f21e993c
# Full Hash: dda3f21e993cf435acc92686243d42f1fa6f5e1c
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-10 09:29:34
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - ContentSubtreeIterator #3: Fix a bug in ContentSubtreeIterator where the end container is in the light DOM of a shadow host r=jjaschke
#   
#   When the end container is in light DOM of a shadow host, we stop the
#   iteration after we have finished iterating the shadow root of it.
#   
# ==============================================================================

diff -r 4669801f1342 -r dda3f21e993c dom/base/ContentIterator.h
--- a/dom/base/ContentIterator.h	Fri May 09 17:16:37 2025 +0000
+++ b/dom/base/ContentIterator.h	Fri May 09 17:16:37 2025 +0000
@@ -121,6 +121,7 @@
       nsINode* aNode,
       dom::AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
           dom::AllowRangeCrossShadowBoundary::No,
+      nsTArray<AncestorInfo>* aInclusiveAncestorsOfEndContainer = nullptr);
   static nsIContent* GetPrevSibling(
       nsINode* aNode,
       dom::AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary =
